fluent-syntax/src/parser.js

```diff
@@ -416,34 +415,37 @@ export default class FluentParser {
   getValue(ps) {
     if (ps.currentIs("{")) {
       ps.peek();
-      ps.peekInlineWS();
-      if (ps.isPeekNextLineVariantStart()) {
+      ps.peekBlank();
+      if (ps.currentPeek() === "*" || ps.currentPeek() === "[") {
+        ps.resetPeek();
         return this.getVariantList(ps);
       }
+
+      ps.resetPeek();
     }
 
     return this.getPattern(ps);
   }
 
   getVariantList(ps) {
     ps.expectChar("{");
-    ps.skipInlineWS();
+    ps.skipBlank();
     const variants = this.getVariants(ps);
-    ps.expectIndent();
+    ps.skipBlank();
     ps.expectChar("}");
     return new AST.VariantList(variants);
   }
 
   getPattern(ps) {
     const elements = [];
-    ps.skipInlineWS();
+    ps.skipBlankInline();
 
     let ch;
     while ((ch = ps.current())) {
 
       // The end condition for getPattern's while loop is a newline
       // which is not followed by a valid pattern continuation.
-      if (ch === "\n" && !ps.isPeekNextLineValue()) {
+      if (ch === "\n" && !ps.isPeekNextLineValue(false)) {
         break;
```

Comment:
We shouldn't break here before first checking if there's an unindented placeable on the next line. The following snippets are valid Fluent 0.7:

```properties
key1 =
{ placeable}

key2 =
{ $selector ->
*[one] One
}
```

The reference parser has a number of fixtures for testing this in [select_indent.ftl](https://github.com/projectfluent/fluent/blob/master/test/fixtures/select_indent.ftl) and [variants_indent](https://github.com/projectfluent/fluent/blob/master/test/fixtures/variants_indent.ftl).

Reply:
This might require a different logic than the one provided by `isPeekNextLineValue`. If so, my other comment about parametrizing `{skip: true}` might not hold anymore. I'd still rename `isPeekSomething` to `isSomething`, though.